src/qililab/platform/platform.py

```diff
@@ -837,38 +844,56 @@ def _execute_qprogram_with_quantum_machines(  # pylint: disable=too-many-locals,
         )
 
-        try:
-            cluster.append_configuration(configuration=configuration)
+        start_time = datetime.datetime.now()
+        for iteration in np.arange(dataloss_tries):  # TODO: This is a temporal fix as QM fixes the dataloss error
+            try:
+                cluster.append_configuration(configuration=configuration)
 
-            if debug:
-                with open("debug_qm_execution.py", "w", encoding="utf-8") as sourceFile:
-                    print(generate_qua_script(qua_program, cluster.config), file=sourceFile)
+                if debug:
+                    with open("debug_qm_execution.py", "w", encoding="utf-8") as sourceFile:
+                        print(generate_qua_script(qua_program, cluster.config), file=sourceFile)
 
-            compiled_program_id = cluster.compile(program=qua_program)
-            job = cluster.run_compiled_program(compiled_program_id=compiled_program_id)
+                compiled_program_id = cluster.compile(program=qua_program)
+                job = cluster.run_compiled_program(compiled_program_id=compiled_program_id)
 
-            acquisitions = cluster.get_acquisitions(job=job)
+                acquisitions = cluster.get_acquisitions(job=job)
 
-            results = QProgramResults()
-            # Doing manual classification of results as QM does not return thresholded values like Qblox
-            for measurement in measurements:
-                measurement_result = QuantumMachinesMeasurementResult(
-                    *[acquisitions[handle] for handle in measurement.result_handles],
+                results = QProgramResults()
+                # Doing manual classification of results as QM does not return thresholded values like Qblox
+                for measurement in measurements:
+                    measurement_result = QuantumMachinesMeasurementResult(
+                        *[acquisitions[handle] for handle in measurement.result_handles],
+                    )
+                    measurement_result.set_classification_threshold(thresholds.get(measurement.bus, None))
+                    results.append_result(bus=measurement.bus, result=measurement_result)
+
+                return results
+
+            except StreamProcessingDataLossError as dataloss:
+                time_interval = datetime.datetime.now() - start_time
+                start_time = datetime.datetime.now()
+                warnings.warn(
+                    f"Warning: {dataloss} raised, retrying experiment ({iteration+1}/{dataloss_tries} available tries) after {time_interval.seconds} s"
                 )
-                measurement_result.set_classification_threshold(thresholds.get(measurement.bus, None))
-                results.append_result(bus=measurement.bus, result=measurement_result)
+                print(traceback.format_exc())
+                if iteration + 1 != dataloss_tries:
+                    time.sleep(1 * dataloss_tries)
```

Comment:
The way it's implemented, this `sleep(dataloss_tries)` timeout will be part of the `time_interval`.

Not wrong, just take in mind that the real `time_interval` would be a bit shorter.

Reply:
Fixed, I moved the time_start after the wait